core/src/test/java/com/gentics/mesh/core/field/s3binary/S3BinaryFieldEndpointTest.java

```diff
@@ -0,0 +1,269 @@
+        //uploading
+        File tempFile = createTempFile();
+        s3BinaryStorage().createBucket("test-bucket").blockingGet();
+        s3BinaryStorage().createBucket("test-cache-bucket").blockingGet();
+        s3BinaryStorage().uploadFile("test-bucket", s3binaryNode.getUuid() + "/s3", tempFile, false).blockingGet();
+        NodeResponse s3 = call(() -> client().extractMetadataNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", metadataBody));
+        assertEquals("test.jpg", s3.getFields().getS3BinaryField(FIELD_NAME).getFileName());
+        assertEquals("image/jpeg", s3.getFields().getS3BinaryField(FIELD_NAME).getMimeType());
+        assertEquals(1376, s3.getFields().getS3BinaryField(FIELD_NAME).getHeight().intValue());
+    }
+
+    @Test
+    public void testExtractMetadataNoFileUploadedShouldFail() {
+        S3RestResponse s3RestResponse = new S3RestResponse();
+        s3RestResponse.setVersion("1");
+        NodeResponse s3binaryNode = createNodeWithField();
+        //creating
+        call(() -> client().updateNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", uploadBody));
+        //uploading
+        File tempFile = createTempFile();
+        s3BinaryStorage().createBucket("test-bucket").blockingGet();
+        s3BinaryStorage().createBucket("test-cache-bucket").blockingGet();
+        //here the call should fail since there is no file uploaded.
+        try {
+            client().extractMetadataNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", metadataBody).blockingGet();
+        } catch (Exception ex) {
+            assertTrue(ex instanceof RuntimeException);
+        }
+    }
+
+    @Test
+    public void testDownloadBinaryWithParams() {
+        S3RestResponse s3RestResponse = new S3RestResponse();
+        s3RestResponse.setVersion("1");
+        NodeResponse s3binaryNode = createNodeWithField();
+        BufferedImage buf = null;
+        //creating
+        call(() -> client().updateNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", uploadBody));
+        //uploading
+        File tempFile = createTempFile();
+        s3BinaryStorage().createBucket("test-bucket").blockingGet();
+        s3BinaryStorage().createBucket("test-cache-bucket").blockingGet();
+        s3BinaryStorage().uploadFile("test-bucket", s3binaryNode.getUuid() + "/s3", tempFile, false).blockingGet();
+        s3BinaryStorage().exists("test-bucket", s3binaryNode.getUuid() + "/s3").blockingGet();
+
+        //extracting metadata in order to resize img
+        call(() -> client().extractMetadataNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", metadataBody));
+
+        // 2. Download the data using the REST API
+        MeshBinaryResponse response = call(() -> client().downloadBinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "en", "s3",
+                new ImageManipulationParametersImpl().setWidth(100)));
+        try {
```

Comment:
I'd use `try-with-resources` here - there would be no need to `fail()` explicitly, as well as close the response.